crash report

Environment
======
Environment Key | Value
:----: | :----:
version | 3.0.2
build | 106
UDID | 8e3c9ff4f43a1c34
Android Release | 4.4.4
Android SDK | 19
Brand | Dell
Device | Venue7
Stack trace
======
```java
java.lang.NullPointerException
	at com.door43.translationstudio.core.LibraryUpdates.removeSourceLanguageUpdate(LibraryUpdates.java:101)
	at com.door43.translationstudio.core.Library.downloadUpdates(Library.java:779)
	at com.door43.translationstudio.tasks.DownloadUpdatesTask.start(DownloadUpdatesTask.java:27)
	at com.door43.util.tasks.ManagedTask.run(ManagedTask.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:841)

```
Log history
======
```java
12/10/15 8:42 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from 
12/10/15 8:42 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from 
12/10/15 4:41 PM  E/com.door43.translationstudio.service.BackupService: Failed to export the target translation uw-obs-sgj
java.io.FileNotFoundException: /data/data/com.translationstudio.androidapp/files/translations/uw-obs-sgj/.git/index.lock: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:409)
	at java.io.FileInputStream.<init>(FileInputStream.java:78)
	at java.io.FileInputStream.<init>(FileInputStream.java:105)
	at com.door43.util.Zip.zipSubFolder(Zip.java:215)
	at com.door43.util.Zip.zipSubFolder(Zip.java:210)
	at com.door43.util.Zip.zip(Zip.java:140)
	at com.door43.translationstudio.core.Translator.exportArchive(Translator.java:203)
	at com.door43.translationstudio.service.BackupService.runBackup(BackupService.java:137)
	at com.door43.translationstudio.service.BackupService.access$100(BackupService.java:33)
	at com.door43.translationstudio.service.BackupService$1.run(BackupService.java:65)
	at java.util.Timer$TimerImpl.run(Timer.java:284)
Caused by: libcore.io.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Posix.open(Native Method)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:110)
	at libcore.io.IoBridge.open(IoBridge.java:393)
	... 10 more

12/10/15 3:44 PM  E/com.door43.translationstudio.service.BackupService: Failed to export the target translation uw-obs-sgj
java.io.FileNotFoundException: /data/data/com.translationstudio.androidapp/files/translations/uw-obs-sgj/.git/index.lock: open failed: ENOENT (No such file or directory)
	at libcore.io.IoBridge.open(IoBridge.java:409)
	at java.io.FileInputStream.<init>(FileInputStream.java:78)
	at java.io.FileInputStream.<init>(FileInputStream.java:105)
	at com.door43.util.Zip.zipSubFolder(Zip.java:215)
	at com.door43.util.Zip.zipSubFolder(Zip.java:210)
	at com.door43.util.Zip.zip(Zip.java:140)
	at com.door43.translationstudio.core.Translator.exportArchive(Translator.java:203)
	at com.door43.translationstudio.service.BackupService.runBackup(BackupService.java:137)
	at com.door43.translationstudio.service.BackupService.access$100(BackupService.java:33)
	at com.door43.translationstudio.service.BackupService$1.run(BackupService.java:65)
	at java.util.Timer$TimerImpl.run(Timer.java:284)
Caused by: libcore.io.ErrnoException: open failed: ENOENT (No such file or directory)
	at libcore.io.Posix.open(Native Method)
	at libcore.io.BlockGuardOs.open(BlockGuardOs.java:110)
	at libcore.io.IoBridge.open(IoBridge.java:393)
	... 10 more

12/10/15 3:14 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from 
12/10/15 3:14 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from 
12/10/15 3:04 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from https://api.unfoldingword.org/obs/txt/1/hi/CQ-hi.json
12/10/15 3:04 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from https://api.unfoldingword.org/obs/txt/1/hi/tN-hi.json
12/10/15 3:04 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from https://api.unfoldingword.org/obs/txt/1/hi/tw_cat-hi.json
12/10/15 3:04 PM  W/com.door43.translationstudio.core.Downloader: Failed to fetch the catalog from https://api.unfoldingword.org/obs/txt/1/hi/kt-hi.json
     
```

--- a/app/src/main/java/com/door43/translationstudio/core/LibraryUpdates.java
+++ b/app/src/main/java/com/door43/translationstudio/core/LibraryUpdates.java
@@ -98,10 +98,12 @@ public class LibraryUpdates implements Serializable {
      */
     public void removeSourceLanguageUpdate(String projectId, String sourceLanguageId) {
         if(mUpdates.containsKey(projectId)) {
-            numSourceTranslationUpdates -= mUpdates.get(projectId).get(sourceLanguageId).size();
-            mUpdates.get(projectId).remove(sourceLanguageId);
-            if(mUpdates.get(projectId).size() == 0) {
-                mUpdates.remove(projectId);
+            if(mUpdates.get(projectId) != null && mUpdates.get(projectId).get(sourceLanguageId) != null) {
+                numSourceTranslationUpdates -= mUpdates.get(projectId).get(sourceLanguageId).size();
+                mUpdates.get(projectId).remove(sourceLanguageId);
+                if (mUpdates.get(projectId).size() == 0) {
+                    mUpdates.remove(projectId);
+                }
             }
         }
     }

